libs/api/database/src/lib/content/stores/tags.store.ts

```diff
@@ -0,0 +1,167 @@
+import { BadRequestException, Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { TagsDocument } from '../schemas';
+import { TagKind, TagsForm } from '@dragonfish/shared/models/content';
+import * as sanitize from 'sanitize-html';
+import { isNullOrUndefined } from '@dragonfish/shared/functions';
+import { TagsTree } from '@dragonfish/shared/models/content/tags.model';
+
+@Injectable()
+export class TagsStore {    
+    private readonly logger: Logger = new Logger(TagsStore.name);
+
+    constructor(
+        @InjectModel('Tags') private readonly tags: Model<TagsDocument>,
+    ) {}
+
+    /**
+     * Returns all tags of a specified kind, sorted alphabetically.
+     * @param kind
+     */
+    async fetchTags(kind: TagKind): Promise<TagsDocument[]> {
+        return this.tags.find({ kind: kind }).sort({ name: 1 });
+    }
+
+    /**
+     * Returns a single tag with the given ID, or null if not found.
+     * @param id The ID of the tag to find.
+     * @returns The tag with the ID.
+     */
+    async findOne(id: string): Promise<TagsDocument> {
+        return this.tags.findById(id);
+    }
+
+    /**
+     * Returns the given tag and all its descendants.
+     * @param king
+     * @param tagId
+     */
+    async fetchDescendants(tagId: string): Promise<TagsTree> {
```

Comment:
A note: getting a list of all _descendants_ will be relatively slow, but I don't expect a tag to ever have too many descendants. Even a few hundred should be fine, especially if we stick a caching layer on the controller, or in the service, or somewhere.

Reply:
lgtm